Brooks/Example/HelloWorld.hs

```diff
@@ -0,0 +1,151 @@
+  (valState $ listPopulateAccess (var "boringList") (listPopulateBool (litInt 5))),
+  (printLnList bool (var "boringList")),
+  (listDec "mySlicedList" 2 $ floatListType static)]
+
+helloListSlice :: (RenderSym repr) => repr (Block repr)
+helloListSlice = (listSlice (floatListType static) (var "mySlicedList") (var "myOtherList") (Just (litInt 1)) (Just (litInt 3)) Nothing)
+
+helloIfBody :: (RenderSym repr) => repr (Body repr)
+helloIfBody = addComments "If body" (body [
+  block [
+    varDec "c" int,
+    varDec "d" int,
+    assign (var "a") (litInt 5),
+    (var "b") &= ((var "a") #+ (litInt 2)),
+    "c" &.= ((var "b") #+ (litInt 3)),
+    (var "d") &=. "b",
+    (var "d") &-= (var "a"),
+    "c" &.-= (var "d"),
+    (var "b") &+= (litInt 17),
+    "c" &.+= (litInt 17),
+    (&++) (var "a"),
+    (&.++) "d",
+    (&~-) (var "c"),
+    (&.~-) "b",
+
+    listDec "myList" 5 (intListType static),
+    objDecDef "myObj" char (litChar 'o'),
+    constDecDef "myConst" string (litString "Imconstant"),
+
+    printLn (int) (var "a"),
+    printLn (int) (var "b"),
+    printLn (int) (var "c"),
+    printLn (int) (var "d"),
+    printLnList (float) (var "myOtherList"),
+    printLnList (float) (var "mySlicedList"),
+    
+    printStrLn "Type an int",
+    getIntInput (var "d"),
+    printStrLn "Type another",
+    discardInput],
+  
+  block [
+    printLn (string) (litString " too"),
+    printStr "boo",
+    print (bool) litTrue,
+    printLn (float) defaultFloat,
+    print (int) (litInt 0),
+    print (char) (litChar 'c'),
+    printLn (bool) ((?!) litTrue),
+    printLn (int) ((#~) (litInt 1)),
+    printLn (float) ((#/^) (litFloat 4.0)),
+    printLn (int) ((#|) (litInt (-4))),
+    printLn (float) (log (litFloat 2.0)),
+    multi [printLn (float) (ln (litFloat 2.0)),
+    printLn (float) (exp ((#~) (litFloat 2.0))),
+    printLn (float) (sin (litFloat 2.0)),
+    printLn (float) (cos (litFloat 2.0)),
+    printLn (float) (tan (litFloat 2.0))],
+    printLn (float) (tan (litFloat 2.0)),
+    printLn (bool) (litTrue ?&& litFalse),
+    printLn (bool) (litTrue ?|| litFalse),
+    printLn (bool) (litTrue ?&& ((?!) litFalse)),
+    printLn (bool) ((?!) (litTrue ?&& litTrue)),
+    printLn (int) ((litInt 6) #+ (litInt 2)),
+    printLn (int) ((litInt 6) #- (litInt 2)),
+    printLn (int) ((litInt 6) #* (litInt 2)),
+    printLn (int) ((litInt 6) #/ (litInt 2)),
+    printLn (int) ((litInt 6) #% (litInt 4)),
+    printLn (int) ((litInt 6) #^ (litInt 2)),
+    printLn (int) ((litInt 6) #+ ((litInt 2) #* (litInt 3))),
+    printLn (float) (csc (litFloat 1.0)),
+    printLn (float) (sec (litFloat 1.0)),
+    printLn (int) (var "a"),
+    printLn (int) (inlineIf (litTrue) (litInt 5) (litInt 0)),
+    printLn (float) (cot (litFloat 1.0))]])
+
+helloElseBody :: (RenderSym repr) => repr (Body repr)
+helloElseBody = bodyStatements [
+  printStrLn "These will print but (by intention) not compile"]
+  -- printLn (int) (notNull (var "a")),
```

Comment:
I don't like having this commented out. It should either be proper code or completely deleted.